### PR TITLE
New package: fractal-3.30.0

### PR DESCRIPTION
**Pull request by CameronNemo**

### CODE REVIEW

**Comment by CameronNemo** [comment]
Cross compilation is not expected to work, since the app is written in Rust. The musl build is not working. The CI builds for x86_64/i686 timed out, but x86_64 succeeded here (it was just really long).

**Comment by jnbr** [comment]
Set `nocross=rust` and add sth. like:
```
case "$XBPS_TARGET_MACHINE" in
    *-musl) broken="reason why broken or link to buildlog" ;;
esac
```
or fix the musl build.

Also, run xlint on the template and fix accordingly.
It would be good, if the build could be slightly more verbose.

File: srcpkgs/fractal/template [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+pkgname=fractal
+version=3.29.6
+revision=1
+build_style=meson
+hostmakedepends="rust cargo pkg-config"
+makedepends="gtk+3-devel glib-devel libressl-devel gstreamer1-devel cairo-devel gettext-devel gst-plugins-base1-devel gst-plugins-bad1-devel gtksourceview-devel pango-devel gspell-devel gmp-devel"
+short_desc="Matrix Messaging Client for GNOME"
+maintainer="Cameron Nemo <camerontnorman@gmail.com>"
+license="GPL-3.0-or-later"
+homepage="https://gitlab.gnome.org/World/fractal/"
```
Comment (maxice8) [inline srcpkgs/fractal/template]: shouldn't it be https://wiki.gnome.org/Apps/Fractal ?

**Comment by CameronNemo** [comment]
The build log problem is marked as wontfix :/

https://gitlab.gnome.org/World/fractal/issues/309

File: srcpkgs/fractal/template [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'fractal'
+pkgname=fractal
+version=3.29.6
+revision=1
+build_style=meson
+hostmakedepends="rust cargo pkg-config"
+makedepends="gtk+3-devel glib-devel libressl-devel gstreamer1-devel cairo-devel gettext-devel gst-plugins-base1-devel gst-plugins-bad1-devel gtksourceview-devel pango-devel gspell-devel gmp-devel"
```
Comment (jnbr) [inline srcpkgs/fractal/template]: please wrap this line at about 80 chars

File: srcpkgs/fractal/template [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'fractal'
+pkgname=fractal
+version=3.29.6
+revision=1
+build_style=meson
+hostmakedepends="rust cargo pkg-config"
+makedepends="gtk+3-devel glib-devel libressl-devel gstreamer1-devel
+cairo-devel gettext-devel gst-plugins-base1-devel gst-plugins-bad1-devel
+gtksourceview-devel pango-devel gspell-devel gmp-devel"
```
Comment (jnbr) [inline srcpkgs/fractal/template]: please indent wrapped lines with a single space

**Comment by jnbr** [comment]
musl fix:
```
pre_build () {
	export GETTEXT_BIN_DIR=/usr/bin
	export GETTEXT_LIB_DIR="${XBPS_CROSS_BASE}/usr/lib/gettext"
	export GETTEXT_INCLUDE_DIR="${XBPS_CROSS_BASE}/usr/include"
}
```

**Comment by CameronNemo** [comment]
thanks for the musl fix! as a heads up, I got a lint error because there was a space between the parentheses and function name.

**Comment by Cogitri** [comment]
Works for me on x86_64, thanks

**Comment by Cogitri** [comment]
https://gitlab.gnome.org/World/fractal/tags/3.30.0

**Comment by CameronNemo** [comment]
Any word on this? Travis fails because of a timeout and there is no way to get around that. It builds successfully locally.

**Comment by Cogitri** [comment]
wfm on x86_64-musl. I guess in _theory_ this could wait until rust cross is working (I'll most likely look into it this weekend) - but there really isn't a need to do so, we can just unbreak it then.

**Comment by jnbr** [comment]
I will merge when the build servers are free. They have a similar timeout, not sure if it will work.

**Comment by maxice8** [comment]
On Wed, Sep 26, 2018 at 01:12:41PM -0700, Johannes wrote:
> I will merge when the build servers are free. They have a similar timeout, not sure if it will work.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/1781#issuecomment-424853701

I plan on merging chromium shortly which will block the builders for quite a while, want to queue yours before ?

**Comment by jnbr** [comment]
go ahead with chromium, will do this one afterwards

**Pull request merged**